### PR TITLE
Descriptive analysis

### PR DESCRIPTION
**Pull request by mdp21oe**

### CODE REVIEW

**Comment by mdp21oe** [comment]
Descriptive stats between benign and malignant nodules

File: index.qmd [hunk 1]
```diff
@@ -157,6 +157,112 @@ Blurb on what Support Vector Machines are.
...
+                  "exposure_radiation",
+                  "final_pathology",
+                  )) |>
+    gtsummary::tbl_summary(by = final_pathology) |>
+  gtsummary::modify_caption("Clinical characteristics between benign and malignant thyrioid nodules")
+clinical_charac
+print(colnames(clinical_charac))
+```
+
+@clinical_charac shows the destribution of clinical variables evaluated between benign and malignant thyroid nodules.
```
Comment (ns-rse) [inline index.qmd:209]: Tpyo...

```suggestion
@clinical_charac shows the distribution of clinical variables evaluated between benign and malignant thyroid nodules.
```

File: index.qmd [hunk 2]
```diff
@@ -157,6 +157,112 @@ Blurb on what Support Vector Machines are.
...
+  gtsummary::modify_caption("Demographics of study population")
+patient_demo
+print(colnames(patient_demo))
+```
+
+@patient_demo shows the demographics of patients included in this study. A total of `r n_obs` patients were included in
+this study with a median (IQR) age of  `r gtsummary::inline_text(patient_demo, variable="age_at_scan")`.
+
+```{r}
+#| label: clinical_charac
```
Comment (ns-rse) [inline index.qmd:184]: No harm in using full words, most IDE's, including RStudio automatically recognise when you are typing references and allow you to hit `TAB` to auto-complete, saving you typing.

```suggestion
#| label: clinical_characteristics
```

Same is true for other section `label`.

Abbreviations and acronyms place a burden/over-head on the reader to decipher them, whilst you or I may be familiar with common usages others may not be, so it helps with readability to be explicit.

File: index.qmd [hunk 3]
```diff
@@ -157,6 +157,112 @@ Blurb on what Support Vector Machines are.
...
+#| eval: true
+#| echo: false
+#| warning: false
+#| tbl-caption: ""
```
Comment (ns-rse) [inline index.qmd:188]: Not sure if the caption defined below in `gtsummary::modify_caption()` is showing for you but if not you could copy and paste it here.

**Review by ns-rse — CHANGES_REQUESTED** [review]
Good start @mdp21oe I noticed on tpyo and have made a few other minor suggestions.

**Comment by mdp21oe** [comment]
Thanks Ns-Rse/Sheffield-Thyroid,


Will make these changes and keep cracking on further.... been reading about
ridge regression, lasso regression and cross validation - some knowledge is
helping to demystify things ha.

BW
Ovie

**Review by mdp21oe — COMMENTED** [review]
Changes made

File: index.qmd [hunk 4]
```diff
@@ -106,24 +106,87 @@ source("r/shf_thy_nod.R")
...
+with thyroid nodules. Data on the following variables were collected: patient demographics (age, gender, ethnicity),
+nodule presentation (incidental nodule, palpable nodule, rapid enlargement, compressive symptoms, vocal paresis), past
+medical history (hypertension, Graves' disease, Hashimotos' thyroiditis, family history of thyroid cancer, exposure to
+neck radiation), biochemistry (thyroid stimulating hormone, lymphocytes, monocytes), ultrasound characteristics (British
+Thyroid Association ultrasound (BTA U), nodule size, solitary nodule, nodule consistency, cervical lymphadenopathy),
+Royal College of Pathology (RCP) FNAC  classification, type of thyroid surgery, and histological diagnosis.
+
+### Data source
+
+Data was collected from patients' case notes and electronic patients' database. Data was collected using a standardised
```
Comment (ns-rse) [inline index.qmd:152]: Can avoid repetition with...

```suggestion
Data was collected from patients' case notes and electronic patients' database using a standardised
```

File: index.qmd [hunk 5]
```diff
@@ -106,24 +106,87 @@ source("r/shf_thy_nod.R")
...
 
-Description of the different models and how they are assessed can go here.
-
-A selection of modern statistical inference approaches have been selected and tested for this work. To fit the models
-data is split into training and testing cohorts in a ratio of `r train`:`r test` and each model is fitted using the
-training cohort. The predictive accuracy of the fitted model is then assessed in the test cohort. **TODO** Expand on this.
+We used a selection of statistic modelling techniques to evaluate association between variables and thyroid cancer in patients with thyroid nodules.
+We used the training and test methodology to split the patient population into training and testing cohorts in a ratio
+of `r train`:`r test` and each model is fitted using the
+training cohort. This split ratio is generally used in traditional machine learning techniques. The training set of the
```
Comment (ns-rse) [inline index.qmd:175]: Ratio's vary on preference `70:30` for some others `80:20` no hard and fast rules for these, the key thing is that you can't use your whole data set to develop your model and assess its accuracy.

File: index.qmd [hunk 6]
```diff
@@ -106,24 +106,87 @@ source("r/shf_thy_nod.R")
...
+data was used to estimate the relation between variables and thyroid cancer. The larger the training data, the better
+it is for the model to learn the trends. The test set was used to determine the accuracy of the model in predicting
+thyroid cancer; the bigger the test data the more confidence we have in the model prognostic values. We used
+randomisation technique for the split to prevent bias in the data split. We ensured that there was no duplicate in the
+data sets so any test data was not accidentally trained.
+Furthermore, cross validation was used to estimate the accuracy of the various machine learning models.
+The k-fold techniques splits the data in ?10 folds, and the data was trained on all but one of the the fold, and the
```
Comment (ns-rse) [inline index.qmd:182]: Number of folds is configurable, 10 is a low and easy number to go with that trades off speed with accuracy, may want to increase when running full analyses.

File: index.qmd [hunk 7]
```diff
@@ -106,24 +106,87 @@ source("r/shf_thy_nod.R")
...
+one fold not trained is used to test the data. This was repeated multiple times using a different fold for test and
+the others for training until all the folds is utilised for training and testing. Following multiple training process
+with k-fold, we selected the model that has the best predictive value for thyroid cancer in the test cohort.
```
Comment (ns-rse) [inline index.qmd:185]: Would be useful to explain why cross-validation is useful.

Its because it avoids over-fitting to the training dataset.

File: index.qmd [hunk 8]
```diff
@@ -106,24 +106,87 @@ source("r/shf_thy_nod.R")
...
+it is for the model to learn the trends. The test set was used to determine the accuracy of the model in predicting
+thyroid cancer; the bigger the test data the more confidence we have in the model prognostic values. We used
+randomisation technique for the split to prevent bias in the data split. We ensured that there was no duplicate in the
+data sets so any test data was not accidentally trained.
+Furthermore, cross validation was used to estimate the accuracy of the various machine learning models.
+The k-fold techniques splits the data in ?10 folds, and the data was trained on all but one of the the fold, and the
+one fold not trained is used to test the data. This was repeated multiple times using a different fold for test and
+the others for training until all the folds is utilised for training and testing. Following multiple training process
+with k-fold, we selected the model that has the best predictive value for thyroid cancer in the test cohort. 
+We also used the leave one out (loo) cross-validation to train and test the data set.In this technique, all but one
```
Comment (ns-rse) [inline index.qmd:186]: I've set code up to allow both, which to "use" is perhaps something yet to be decided, but its possible to write that two methods of cross-validation were assessed in the methods and report on the results of both and make a decision as to which to report overall, explaining why.

The why might be decided by comparing how much variance (i.e. how close each of the folds or leave-one-out) there is and going with the method that provided the most robust (lowest variance) estimates.

File: index.qmd [hunk 9]
```diff
@@ -140,8 +203,8 @@ variables @steyerberg2001
 
 
 #### Random Forest
+The random forest plot is an extension of the decision tree methodology to reduce the variance. Decision trees are very sensitive to the training data set and can lead to high variance; thus potential issues with generalisation of the model. The random forest plot selects random observation of the data set to create multiple decision trees. Random variables are selected for each tree to use in the training of the data set. The aggregated output of the generated decision trees is then used to create an estimate. The random forest methology is associated with reduced variance.
```
Comment (ns-rse) [inline index.qmd:206]: Random forest isn't a plot, although you can plot individual forests that are generated by random forests.

The point of Random Forests is that they avoid over-fitting which can arise when you use a single decision tree, since as you've clocked random samples are selected from the training data set and a decision tree fitted to each.

It would be worth explaining to readers how decision trees work with classification in that they select one of the predictor variables that optimally splits the cohort into the outcome of interest (which in this case is whether a patient has malignant or benign tumors). This is then repeated in each of the subsets using another of the predictors and the process is repeated until the user chosen "depth" of the tree is found.

Random Forests take this ensemble of decision trees made on subsets of the training data and then aggregate the results.

**Review by ns-rse — APPROVED** [review]
Looks good, lets merge this and use new branches for subsequent work. :+1:
